signed/verifiers.go

```diff
@@ -0,0 +1,89 @@
+package signed
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"log"
+	"reflect"
+
+	"github.com/flynn/go-tuf/Godeps/_workspace/src/github.com/agl/ed25519"
+)
+
+type Verifier interface {
+	Verify(key []byte, msg []byte, sig []byte) error
+}
+
+// Verifiers serves as a map of all verifiers available on the system and
+// can be injected into a verificationService. For testing and configuration
+// purposes, it will not be used by default.
+var Verifiers = map[string]Verifier{
+	"ed25519": Ed25519Verifier{},
+	//"rsa":     RSAVerifier{},
+}
+
+// RegisterVerifier provides a convenience function for init() functions
+// to register additional verifiers or replace existing ones.
+func RegisterVerifier(name string, v Verifier) {
+	curr, ok := Verifiers[name]
+	if ok {
+		typOld := reflect.TypeOf(curr)
+		typNew := reflect.TypeOf(v)
+		log.Printf(
+			"Replacing already loaded verifier %s:%s with %s:%s\n",
```

Comment:
No problem. Docker is mostly using logrus which lets us control log level, so I've been moving all our random bits of logging to that but I didn't want to add that dependency for your fork. I'm happy to just take it all out.